### PR TITLE
CAFV-357: Add new v1beta3 API to CAPVCD

### PR DESCRIPTION
**Pull request by arunmk**
## Description
Please provide a brief description of the changes proposed in this Pull Request

- Added new API version in anticipation of upcoming changes in CAPVCD and also as a CoDB since the branch of the current release has been cut. 

## Checklist
- [X] tested locally
- [X] updated any relevant dependencies
- [ ] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [X] Yes
- [ ] No

If yes, please fill in the below

1. Updated conversions?
    - [X] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [ ] Yes
    - [ ] No
    - [X] N/A
3. Updated infrastructure-components.yaml?
    - [X] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [X] No
    - [ ] N/A
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [ ] Yes
   - [X] No I did not find v1.0.0 or anything other than v0.5.1
   - [ ] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/550)

### CODE REVIEW

File: api/v1beta1/conversion.go [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+	return autoConvert_v1beta3_VCDMachineTemplateResource_To_v1beta1_VCDMachineTemplateResource(in, out, s)
 }
 
-func Convert_v1beta2_VCDMachineSpec_To_v1beta1_VCDMachineSpec(in *v1beta2.VCDMachineSpec, out *VCDMachineSpec, s conversion.Scope) error {
-	return autoConvert_v1beta2_VCDMachineSpec_To_v1beta1_VCDMachineSpec(in, out, s)
+func Convert_v1beta3_VCDMachineSpec_To_v1beta1_VCDMachineSpec(in *v1beta3.VCDMachineSpec, out *VCDMachineSpec, s conversion.Scope) error {
+	return autoConvert_v1beta3_VCDMachineSpec_To_v1beta1_VCDMachineSpec(in, out, s)
+}
+
+func Convert_v1beta3_VCDClusterSpec_To_v1beta1_VCDClusterSpec(in *v1beta3.VCDClusterSpec, out *VCDClusterSpec, s conversion.Scope) error {
```
Comment (Anirudh9794) [inline api/v1beta1/conversion.go:16]: I am wondering why these functions were in zz_generated_conversions.go for v1beta2

**Review by Anirudh9794 — APPROVED** [review]
I just have a small question. But the changes look good.

**Comment by arunmk** [comment]
@sahithi I have made the changes requested. I will run through the tests soon.

File: api/v1beta1/conversion.go (hunk 1 above)
Comment (arunmk) [reply api/v1beta1/conversion.go:16]: I am not certain of this. There are some functions in the regular zz_generated_conversions, and some are missing.

**Review by sahithi — APPROVED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/550)* status: all files reviewed, 6 unresolved discussions (waiting on @abaruni, @Anirudh9794, and @arunmk)

**Review by Anirudh9794 — APPROVED** [review]
LGTM

**Pull request merged**